src/components/Button/__tests__/button.jest.tsx

```diff
@@ -1,18 +1,25 @@
 import React from "react";
-import { render, fireEvent, cleanup } from "@testing-library/react";
-import Button from "../Button";
+import { render, fireEvent, cleanup, RenderResult } from "@testing-library/react";
+import Button, { ButtonProps } from "../Button";
 
 const text = "Click Me!";
 const className = "test-class";
+const mockOnClick = jest.fn();
```

Comment:
this might be dangerous - the mock's state is never reset between tests. Consider calling `jest.resetAllMocks()` in an `afterEach`, or creating a new mock yourself in a `beforeEach`